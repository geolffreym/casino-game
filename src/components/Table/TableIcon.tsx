import styled from 'styled-components'
import { WithTheme, withTheme } from '@material-ui/core'
import React, { FC, PropsWithChildren } from 'react'

const TableIcon: FC<PropsWithChildren<{ Icon: FC }>> = ({ Icon, children }): JSX.Element => {
  return (
    <TableIconWrapper>
      <IconWrapper>
        <Icon />
      </IconWrapper>
      <TableIconText>
        {children}
      </TableIconText>
    </TableIconWrapper>
  )
}

export default TableIcon

const TableIconWrapper = withTheme(
  styled.div<WithTheme>`
    display: flex;

    svg {
      fill: ${({ theme }) => theme.palette.grey[250]};
    }
  `
)

export const TableIconText = withTheme(
  styled.div<WithTheme>`
    color: ${({ theme }) => theme.palette.grey[250]};
    margin-left: 0.5rem;
  `
)

const IconWrapper = styled.div`
  width: 1rem;
  height: 1rem;
  
  svg {
    width: 100%;
    height: 100%;
  }
`
