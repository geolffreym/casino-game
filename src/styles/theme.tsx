// Global styles can be moved to a separate file for ease of maintenance.
export const globalOverrides = {
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          margin: 0,
          padding: 0,
          height: '100%'
        },
        body: {
          margin: 0,
          padding: 0,
          height: '100%',
          fontSize: '1rem',
          backgroundColor: '#fff'
        }
      }
    }
  }
}

export const typography = {
  typography: {
    fontFamily: [
      'Tamil Sangam MN',
      'Oswald',
      'Arial',
      'sans-serif'
    ].join(',')
  }
}

export const defaultTheme = {
  palette: {
    primary: {
      light: '#bf1f34',
      main: '#990015',
      dark: '#750212',
      contrastText: '#E4F1F0'
    },
    secondary: {
      main: '#2795B7',
      contrastText: '#fff'
    },
    warning: {
      light: '#FEF5B2',
      main: '#C79C36',
      dark: '#814E22'
    },
    success: {
      light: '#DEF7EC',
      main: '#3C763D',
      dark: '#2D6C59'
    },
    default: {
      main: '#fff',
      contrastText: '#9999'
    },
    grey: {
      50: '#F2F6FF',
      150: '#F9FAFC',
      200: '#E6EBF1',
      250: '#98A5BA'
    }
  }
}
