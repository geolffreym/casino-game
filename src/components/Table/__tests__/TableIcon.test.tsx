import TableIcon, { TableIconText } from '@components/Table/TableIcon'
import { Theme } from '@material-ui/core'
import { mount, shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import { Transform } from '@material-ui/icons'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { defaultTheme } from '@styles/theme'

describe('<TableIcon />', () => {
  let theme: Theme

  beforeAll(() => {
    theme = createTheme(defaultTheme)
  })

  it('should render table icon', () => {
    const component = shallow(<TableIcon Icon={Transform}>test</TableIcon>)
    expect(component).toMatchSnapshot()
  })

  it('should have table icon props text', () => {
    const contentText = 'Test Module'
    const component = shallow(<TableIcon Icon={Transform}>{contentText}</TableIcon>)
    const tableIconText = component.find(TableIconText)

    expect(tableIconText.text()).toEqual(contentText)
  })

  it('should have table icon text valid theme styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <TableIconText />
      </ThemeProvider>
    )

    // @ts-expect-error
    expect(component).toHaveStyleRule('color', theme.palette.grey[250])
  })
})
