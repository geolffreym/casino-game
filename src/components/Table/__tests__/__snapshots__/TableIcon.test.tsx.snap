// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableIcon /> should render table icon 1`] = `ShallowWrapper {}`;
